src/main/java/com/exadel/sandbox/employee/service/impl/EmployeeServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package com.exadel.sandbox.employee.service.impl;
+
+import com.exadel.sandbox.employee.dto.employeeDto.EmployeeCreateDto;
+import com.exadel.sandbox.employee.dto.employeeDto.EmployeeResponseDto;
+import com.exadel.sandbox.employee.dto.employeeDto.EmployeeUpdateDto;
+import com.exadel.sandbox.employee.dto.tgInfoDto.TgInfoResponseDto;
+import com.exadel.sandbox.employee.entity.TgInfo;
+import com.exadel.sandbox.employee.service.EmployeeService;
+import com.exadel.sandbox.exception.EntityNotFoundException;
+import com.exadel.sandbox.employee.entity.Employee;
+import com.exadel.sandbox.employee.repository.EmployeeRepository;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Component;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class EmployeeServiceImpl implements EmployeeService {
+
+    private final EmployeeRepository employeeRepository;
+    private final ModelMapper mapper;
+
+    @Override
+    @Transactional
+    public List<EmployeeResponseDto> getEmployees() {
+        List<Employee> employees = employeeRepository.findAll();
+
+        List<EmployeeResponseDto> employeeResponseDtos = new ArrayList<>();
+
+        for (Employee employee: employees) {
+            employeeResponseDtos.add(fullMap(employee));
+        }
+
+
+        return employeeResponseDtos;
+    }
+
+    @Override
+    @Transactional
+    public EmployeeResponseDto getEmployeeByID(Long id) {
+        Optional<Employee> byId = employeeRepository.findById(id);
+
+        Employee employee = byId.orElseThrow(() -> new EntityNotFoundException("Employee with id: " + id + " not found"));
+
+        return fullMap(employee);
+    }
+
+    @Override
+    @Transactional
+    public EmployeeResponseDto create(EmployeeCreateDto employeeCreateDto) {
+        Employee employee = mapper.map(employeeCreateDto, Employee.class);
+
+        if(employeeCreateDto.getTgInfoResponseDto() != null)
+            employee.setTgInfo(mapper.map(employeeCreateDto.getTgInfoResponseDto(), TgInfo.class));
+
+
+
+        return fullMap(employeeRepository.save(employee));
+    }
+
+    @Override
+    @Transactional
+    public void deleteById(Long id) {
+        employeeRepository.deleteById(id);
+    }
+
+    @Override
+    @Transactional
+    public EmployeeResponseDto update(Long id, EmployeeUpdateDto employeeUpdateDto) {
+        Employee employee = mapper.map(employeeUpdateDto, Employee.class);
```

Comment:
Update method should work like:
1. Get entity using id
2. Update entity using dto and mapper
3. Save changed object to DB